### PR TITLE
chore: Enable Foundry tests

### PR DESCRIPTION
**Pull request by Jrigada**
# What :computer: 
* This PR integrates and enables the Foundry test suite to ensure comprehensive testing of the codebase.

# Why :hand:
* To increase robustness and reliability of the project by leveraging Foundry's testing capabilities.
* To validate new code contributions while ensuring that existing functionality remains intact.

### CODE REVIEW

**Review by Karrq — DISMISSED** [review]

File: crates/cheatcodes/src/inspector.rs [hunk 1]
```diff
@@ -1550,8 +1550,10 @@ where {
             let prev = account.info.nonce;
             let nonce = prev.saturating_sub(1);
             account.info.nonce = nonce;
-            // NOTE(zk): We sync with the nonce changes to ensure that the nonce matches
-            foundry_zksync_core::cheatcodes::set_nonce(sender, U256::from(nonce), ecx_inner);
+            if self.use_zk_vm {
```
[the remark this answers is not in the document]
Comment (Jrigada) [reply crates/cheatcodes/src/inspector.rs:1553]: Yes indeed

File: crates/forge/tests/it/test_helpers.rs [hunk 2]
```diff
@@ -247,6 +247,7 @@ impl ForgeTestData {
     pub fn new(profile: ForgeTestProfile) -> Self {
         init_tracing();
 
+        let _ = rustls::crypto::ring::default_provider().install_default();
```
Comment (nbaztec) [inline crates/forge/tests/it/test_helpers.rs:250]: Please add a comment why we have this

```suggestion
        // NOTE(zk): We need to manually install the crypto provider as zksync-era uses `aws-lc-rs` provider, while
        // foundry uses the `ring` provider. As a result, rustls cannot disambiguate between the two while selecting a default provider.
        let _ = rustls::crypto::ring::default_provider().install_default();
```

File: crates/cast/bin/main.rs [hunk 3]
```diff
@@ -45,6 +45,9 @@ fn main() {
 }
 
 fn run() -> Result<()> {
+    // Initialize crypto provider
```
Comment (nbaztec) [inline crates/cast/bin/main.rs:48]:
```suggestion
    // NOTE(zk): We need to manually install the crypto provider as zksync-era uses `aws-lc-rs` provider, while
    // foundry uses the `ring` provider. As a result, rustls cannot disambiguate between the two while selecting a default provider.
```

File: .github/scripts/matrices.py [hunk 4]
```diff
@@ -71,8 +71,9 @@ def __init__(
 # TODO: Figure out how to make this work
 # t_linux_arm = Target("ubuntu-latest", "aarch64-unknown-linux-gnu", "linux-aarch64")
 t_macos = Target("macos-latest", "aarch64-apple-darwin", "macosx-aarch64")
-t_windows = Target("windows-latest", "x86_64-pc-windows-msvc", "windows-amd64")
-targets = [t_linux_x86, t_windows] if is_pr else [t_linux_x86, t_macos, t_windows]
+# note(ZK): ZKsync-era doesn't support windows as of now
```
Comment (elfedy) [inline .github/scripts/matrices.py:74]:
```suggestion
# NOTE(zk): ZKsync-era doesn't support windows as of now
```

File: crates/config/src/lib.rs [hunk 5]
```diff
@@ -2212,7 +2212,7 @@ impl Default for Config {
             allow_paths: vec![],
             include_paths: vec![],
             force: false,
-            evm_version: EvmVersion::default(),
+            evm_version: EvmVersion::Paris,
```
Comment (elfedy) [inline crates/config/src/lib.rs:2215]: Any reason we default to Paris? what does upstream do?

File: crates/cheatcodes/src/inspector.rs [hunk 6]
```diff
@@ -1550,8 +1550,10 @@ where {
...
+                // NOTE(zk): We sync with the nonce changes to ensure that the nonce matches
+                foundry_zksync_core::cheatcodes::set_nonce(sender, U256::from(nonce), ecx_inner);
```
Comment (elfedy) [inline crates/cheatcodes/src/inspector.rs:1555]: How is this related to enabling the foundry tests?

**Review by elfedy — DISMISSED** [review]

File: crates/config/src/lib.rs (hunk 5 above)
Comment (Karrq) [reply crates/config/src/lib.rs:2215]: upstream defaults to `Paris` actually, we had a regression in defaulting to `default` that I have identified as well

File: crates/cheatcodes/src/inspector.rs (hunk 6 above)
Comment (Karrq) [reply crates/cheatcodes/src/inspector.rs:1555]: because before it was running on _all_ regardless if zkvm was enabled or not

**Review by Karrq — DISMISSED** [review]

File: crates/cheatcodes/src/inspector.rs (hunk 6 above)
Comment (Jrigada) [reply crates/cheatcodes/src/inspector.rs:1555]: We were accessing the nonce_holder_Address even though we were in evm exclusively and that was making an extra storage access that change the dictionary used by the fuzz tester

**Review by elfedy — APPROVED** [review]

**Review by Karrq — APPROVED** [review]
